contracts/stellar-axelar-gas-service/src/contract.rs

```diff
@@ -40,7 +40,7 @@ impl AxelarGasServiceInterface for AxelarGasService {
     ) -> Result<(), ContractError> {
         spender.require_auth();
 
-        ensure!(token.amount > 0, ContractError::InvalidAmount);
+        ensure!(token.amount >= 0, ContractError::InvalidAmount);
```

Comment:
Make `gas_token` an `Option<Token>` in ITS instead. Same for the example contract. If it's not attached, gas service isn't called. This will mimic the EVM ITS reference behaviour